llvm/include/llvm/Telemetry/Telemetry.h

```diff
@@ -8,71 +8,68 @@
 ///
 /// \file
 /// This file provides the basic framework for Telemetry
-///
-/// It comprises of three important structs/classes:
-///
-/// - Telemeter: The class responsible for collecting and forwarding
-///              telemery data.
-/// - TelemetryInfo: data courier
-/// - TelemetryConfig: this stores configurations on Telemeter.
-///
 /// Refer to its documentation at llvm/docs/Telemetry.rst for more details.
 //===---------------------------------------------------------------------===//
 
 #ifndef LLVM_TELEMETRY_TELEMETRY_H
 #define LLVM_TELEMETRY_TELEMETRY_H
 
-#include <chrono>
-#include <ctime>
-#include <memory>
-#include <optional>
-#include <string>
-
 #include "llvm/ADT/StringExtras.h"
 #include "llvm/ADT/StringRef.h"
 #include "llvm/Support/Error.h"
 #include "llvm/Support/JSON.h"
+#include <memory>
+#include <optional>
+#include <string>
 
 namespace llvm {
 namespace telemetry {
 
+class Serializer {
+public:
+  virtual llvm::Error start() = 0;
```

Comment:
If `start` is intended to be for some sort of delayed initialisation, perhaps worth calling it `init` to be more idiomatic?

Reply:
done